5004 delete step group

X0nic opened the pull request:
There are changes in here from 5005. Use this link to see what was actually changed here. https://github.com/sassafrastech/madeline_system/compare/928b769c5b2b0a482065288f73d27401b1c9ec6d...5004_delete_step_group?expand=1

X0nic commented:
Paired with @linesarefuzzy and now it works as expected.

hooverlunch commented on app/controllers/admin/project_groups_controller.rb:
```diff
@@ -23,14 +34,40 @@ def create
+    if @entry.destroy
+      render nothing: true
+    else
+      render_modal_partial(status: 422)
```
This doesn't make sense to me. Why would we render the modal partial? I don't think we are ever expecting destroy to fail, right? So I would actually raise an error if it does perhaps?

hooverlunch commented on app/views/admin/loans/timeline/_main.slim:
```diff
@@ -19,4 +19,9 @@ section.timeline-table
     li: a href="#" data-action="edit" = t(:edit)
-    li: a href="#" data-action="delete" = t(:delete)
\ No newline at end of file
+    li.disabled: a href="#" Delete
```
Please add a comment explaining why there are two versions here and how they get shown/hidden (via the with-children class), etc. Else this is kind of cryptic.

hooverlunch commented on app/assets/javascripts/backbone/views/timeline_table_view.coffee:
```diff
@@ -9,7 +9,12 @@ class MS.Views.TimelineTableView extends Backbone.View
+    'click #project-group-menu [data-action="add-child-group"]': 'newChildGroup'
     'click .project-group .fa-cog': 'openGroupMenu'
+    'click .project-group .disabled a': 'disableDeleteLink'
+    'click .project-group a[data-action="delete"]': 'disableDeleteLink'
```
These will go away as we discussed.

hooverlunch commented on app/assets/stylesheets/admin/loans/timeline/_table.scss:
```diff
@@ -42,4 +42,26 @@
   }
+
+  // Delete button enabled
+  .project-group .dropdown-menu {
```
Let's be sure to reuse this stuff if we need to be able to disable the delete link for steps also.

hooverlunch commented on app/views/admin/loans/timeline/_group.slim:
```diff
@@ -1,14 +1,15 @@
       = entry.summary
       span.dropdown
-        i.fa.fa-cog data-toggle="dropdown" aria-hidden="true"
+        i.fa.fa-cog data-toggle="dropdown" aria-hidden="true" data-delete-disabled="true"
```
Remove data-delete-disabled?

hooverlunch commented on config/locales/es/es.yml:
```diff
@@ -389,6 +389,7 @@ es:
 
   project_group:
     add: "Translate Me"
+    edit: "Translate Me"
```
These things are not necessary. Our translation system detects missing strings. Thanks though!

X0nic replied:
Should I remove these then? Would this induce a false positive?

X0nic commented on app/assets/javascripts/backbone/views/timeline_table_view.coffee:
```diff
@@ -22,7 +27,37 @@ class MS.Views.TimelineTableView extends Backbone.View
+    e.preventDefault()
+
+    $project_group = $(e.target).closest(".project-group")
+    project_group_id = $project_group.data("action-key")
```
I saw that being used in `batch_actions` and thought I would copy the same name. I do like `id` better, though.

X0nic commented on app/assets/javascripts/backbone/views/timeline_table_view.coffee:
```diff
@@ -22,7 +27,37 @@ class MS.Views.TimelineTableView extends Backbone.View
+    e.preventDefault()
+
+    $project_group = $(e.target).closest(".project-group")
+    project_group_id = $project_group.data("action-key")
```
Yeah, makes sense.

X0nic commented:
This should cover everything except for the extra translation.

I also refactored this to use one delete button with `stopPropagation()`. The duplicate elements are no longer needed. I have removed the CSS that used to be required.

hooverlunch commented:
Yes please remove.

On Tue, Oct 18, 2016 at 12:48 PM, Nathan Lee <notifications@github.com>
wrote:

> *@X0nic* commented on this pull request.
> ------------------------------
>
> In config/locales/es/es.yml
> <https://github.com/sassafrastech/madeline_system/pull/163>:
>
> > @@ -389,6 +389,7 @@ es:
>
>    project_group:
>      add: "Translate Me"
> +    edit: "Translate Me"
>
> Should I remove these then? Would this induce a false positive?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/163>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AAq3cMOLyReKLpWH_mSV4NNNJdCikZXYks5q1PhxgaJpZM4KWdo9>
> .
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech
